cvat-ui/src/components/annotation-page/canvas/canvas-wrapper3D.tsx

```diff
@@ -59,19 +71,98 @@ const CanvasWrapperComponent = (props: Props): ReactElement => {
+        canvasInstanceDOM.perspective.addEventListener('dblclick', onMouseDoubleClick);
+        canvasInstanceDOM.perspective.addEventListener('click', onMouseClick);
     };
 
     const keyControls = (key: KeyboardEvent): void => {
         const { canvasInstance } = props;
         canvasInstance.keyControls(key);
     };
 
+    const onPerspectiveViewResize = (e: MouseEvent): void => {
+        const canvas3dContainer = document.getElementById('canvas3d-container');
+        if (canvas3dContainer) {
+            const height = canvas3dContainer.clientHeight + canvas3dContainer.getBoundingClientRect().top - e.clientY;
+            setOrthographicViewSize({ ...orthographicViewSize, vertical: height });
+        }
+    };
+
+    const onOrthographicViewResize = (view: string, e: MouseEvent): void => {
+        const canvas3dContainer = document.getElementById('canvas3d-container');
+        if (canvas3dContainer) {
+            const width = e.clientX - canvas3dContainer.getBoundingClientRect().left;
+            if (view === ViewType.TOP) {
+                const topWidth = orthographicViewSize.top;
+                if (topWidth < width) {
+                    const top = orthographicViewSize.top + (width - topWidth);
+                    const side = orthographicViewSize.side - (width - topWidth);
+                    setOrthographicViewSize({
+                        ...orthographicViewSize,
+                        top,
+                        side,
+                    });
+                } else {
+                    const top = orthographicViewSize.top - (topWidth - width);
+                    const side = orthographicViewSize.side + (topWidth - width);
+                    setOrthographicViewSize({
+                        ...orthographicViewSize,
+                        top,
+                        side,
+                    });
+                }
+            }
+            if (view === ViewType.SIDE) {
+                const topSideWidth = orthographicViewSize.top + orthographicViewSize.side;
+                if (topSideWidth < width) {
+                    const side = orthographicViewSize.side + (width - topSideWidth);
+                    const front = orthographicViewSize.front - (width - topSideWidth);
+                    setOrthographicViewSize({
+                        ...orthographicViewSize,
+                        side,
+                        front,
+                    });
+                } else {
+                    const side = orthographicViewSize.side - (topSideWidth - width);
+                    const front = orthographicViewSize.front + (topSideWidth - width);
+                    setOrthographicViewSize({
+                        ...orthographicViewSize,
+                        side,
+                        front,
+                    });
+                }
+            }
+        }
+    };
+
     useEffect(() => {
```

Comment:
This useEffect works each component render, not only when component was first time mounted. 
Probably it is a mistake? We can use ``[]`` as dependency argument, in this case useEffect works on mount only.

Reply:
Resolved